astropy/modeling/rotations.py

```diff
@@ -187,9 +187,6 @@ def evaluate(cls, x, y, angle):
         single coordinate pair.
         """
 
-        if x.shape != y.shape:
-            raise ValueError("Expected input arrays to have the same shape")
```

Comment:
That's a little broad-- `Rotation2D` *can* be used in compound models in general.  The problem you identified has less to do with `Rotation2D` and more to do with the implementation of `Mapping` and that it isn't normalizing scalar inputs to match the expectations of how most other models are implemented (all the input normalization that normally happens in `prepare_inputs`).  This is is because in the example you gave there's an `Identity` at the very beginning of the transformation, but the way I implemented `Mapping` assumed that it would always be wedged between two other models, which clearly isn't true (I guess?) so that's what I need to fix.